core/lib/multivm/src/versions/vm_latest/tests/l1_l2_complete_tx_execution.rs

```diff
@@ -0,0 +1,602 @@
+    mailbox::BridgeHubRequestL2TransactionOnGateway,
+    Execute, U256,
+};
+use zksync_utils::u256_to_h256;
+
+use crate::{
+    interface::{ExecutionResult, TxExecutionMode, VmExecutionMode, VmInterface},
+    vm_latest::{
+        tests::{
+            tester::{TxType, VmTester, VmTesterBuilder},
+            utils::{
+                read_admin_facet, read_bridgehub, read_diamond, read_diamond_init,
+                read_diamond_proxy, read_mailbox_facet, read_stm, read_transparent_proxy,
+                verify_required_storage, BASE_SYSTEM_CONTRACTS,
+            },
+        },
+        HistoryEnabled,
+    },
+};
+
+fn send_prank_tx_and_verify(
+    vm: &mut VmTester<HistoryEnabled>,
+    contract_address: Address,
+    factory_deps: Vec<Vec<u8>>,
+    calldata: Vec<u8>,
+    prank_address: Address,
+) {
+    let account: &mut Account = &mut vm.rich_accounts[0];
+
+    let tx = account.get_l1_prank_tx(
+        Execute {
+            contract_address,
+            value: U256::zero(),
+            factory_deps,
+            calldata,
+        },
+        0,
+        prank_address,
+    );
+    vm.vm.push_transaction(tx);
+
+    let res = vm.vm.execute(VmExecutionMode::OneTx);
+
+    match res.result {
+        ExecutionResult::Success { output } => {
+            println!("Transaction was successful. Output: {:?}", output);
+        }
+        ExecutionResult::Revert { output } => {
+            eprintln!("Transaction reverted. Reason: {:?}", output);
+            panic!("Transaction wasn't successful: {:?}", output);
+        }
+        ExecutionResult::Halt { reason } => {
+            eprintln!("Transaction halted. Reason: {:?}", reason);
+            panic!("Transaction wasn't successful: {:?}", reason);
+        }
+    }
+}
+
+fn deploy_and_verify_contract(
+    vm: &mut VmTester<HistoryEnabled>,
+    contract_code: &Vec<u8>,
+    constructor_data: Option<&[Token]>,
+    deploy_nonce: &mut u64,
+) -> Address {
+    let account: &mut Account = &mut vm.rich_accounts[0];
+
+    let deploy_tx = account.get_deploy_tx(
+        contract_code,
+        constructor_data,
+        TxType::L1 {
+            serial_id: *deploy_nonce,
+        },
+    );
+    *deploy_nonce += 1;
+    let contract_address = deploy_tx.address.clone();
+
+    vm.vm.push_transaction(deploy_tx.tx.clone());
+
+    let res = vm.vm.execute(VmExecutionMode::OneTx);
+
+    // The code hash of the deployed contract should be marked as republished.
+    let known_codes_key = get_known_code_key(&deploy_tx.bytecode_hash);
+
+    // The contract should be deployed successfully.
+    let account_code_key = get_code_key(&deploy_tx.address);
+
+    let expected_slots = vec![
+        (u256_to_h256(U256::from(1u32)), known_codes_key),
+        (deploy_tx.bytecode_hash, account_code_key),
+    ];
+    assert!(!res.result.is_failed());
+
+    verify_required_storage(&vm.vm.state, expected_slots);
+
+    contract_address
+}
+
+fn send_l2_tx_and_verify(
```

Comment:
these seem like a commonly used functions, could you please move those to the utils file in the test folder?